contracts/strategies/fluid/FluidStrategy.sol

```diff
@@ -0,0 +1,205 @@
+            uint256 harvestAmt = _splitAndSendReward(_asset, yieldReceiver, msg.sender, assetInterest);
+            emit InterestCollected(_asset, yieldReceiver, harvestAmt);
+        }
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function supportsCollateral(address _asset) external view override returns (bool) {
+        return assetToPToken[_asset] != address(0);
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkBalance(address _asset) external view override returns (uint256 balance) {
+        balance = allocatedAmount[_asset];
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkAvailableBalance(address _asset) external view override returns (uint256) {
+        uint256 availableLiquidity = _getAvailableLiquidity(_asset);
+        uint256 allocatedValue = allocatedAmount[_asset];
+        if (availableLiquidity <= allocatedValue) {
+            return availableLiquidity;
+        }
+        return allocatedValue;
+    }
+
+    /// @notice Collect accumulated reward token and send to Vault.
+    /// @dev There are no separate rewards by Fluid.
+    function collectReward() external pure override {
+        revert NoRewardIncentive();
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkRewardEarned() external pure override returns (RewardData[] memory) {
+        return (new RewardData[](0));
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkInterestEarned(address _asset) public view override returns (uint256 interest) {
+        uint256 availableLiquidity = _getAvailableLiquidity(_asset);
+        uint256 allocatedValue = allocatedAmount[_asset];
+        if (availableLiquidity > allocatedValue) {
+            interest = availableLiquidity - allocatedValue;
+        }
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkLPTokenBalance(address _asset) public view override returns (uint256 balance) {
+        balance = IERC20(_getPTokenFor(_asset)).balanceOf(address(this));
+    }
+
+    /// @notice Internal withdraw function used for withdrawing from the strategy.
+    /// @param _recipient Receiver of the funds.
+    /// @param _asset Asset to be withdrawn.
+    /// @param _amount Amount to be withdrawn.
+    /// @return _amount Amount withdrawn.
+    function _withdraw(address _recipient, address _asset, uint256 _amount) internal returns (uint256) {
+        Helpers._isNonZeroAddr(_recipient);
+        Helpers._isNonZeroAmt(_amount, "Must withdraw something");
+
+        // Checking for shares required to be burned to get the desired _amount and checking maximum redeemable shares.
+        address lpToken = _getPTokenFor(_asset);
+        uint256 shares = IfToken(lpToken).previewWithdraw(_amount);
+        uint256 maxRedeemable = IfToken(lpToken).maxRedeem(address(this));
+        if (shares > maxRedeemable) {
+            revert LimitReached();
```

Comment:
It might make sense to add maxRedeemable here

Reply:
It is added already

Reply:
I meant, something like:
`revert LimitReached(maxRedeemable);`

Reply:
It would be redundant, also now we are not caching the max amounts while deposit and redeem. So that's a good optimisation and this value can anyway easily be fetched.